### DISCUSSION TITLE
Replacing Asset ID after deleting previous Item

### DISCUSSION THREAD

**Discussion by DanCharon** [snipe/snipe-it#10147] [Q&A]
Hi Everyone,

I hope you are all safe and well!

We've encountered an issue with our version of Snipe and I was hoping I could pick the brain of the SnipeIt collective to find a solution. 

In our Snipe install our Asset ID's must match our asset tags (ie. Asset with the ID of 508 is named as ASSET508) as we use a QR code which when scanned takes us directly to the URL of said asset. 

After a series of deletions we are trying to continue from ID 508 so it will line up with the next asset name. (Though we can change the asset name to match it is the ID which is important). 

After purging all of the soft delete records I am still seeing a problem when creating a new asset. Whereby instead of starting from ID 508, it starts much later down the line at the next asset that would have been in line before the deletions.

![Screenshot 2021-09-29 at 17 02 08](https://user-images.githubusercontent.com/91482229/135306630-47664d56-de53-49f0-bc81-11341a9ed5c2.png)

Is there anything else I can do to make snipe recognise that these asset IDs no longer exist and begin from the ones we want? 

Cheers!

**Comment by snipe** [comment]
That's pretty standard. Deleting records doesn't free up their old ID in an euro-incrementing database. The asset ID shouldn't have to match the asset tag for the QR codes to work though - that's kind of the point of them. The Snipe-IT QR code doesn't care what your asset tag is, it just points to the ID URL.

**Comment by uberbrady** [comment]
*Generally* we advise against trying to to do anything with ID's - the database will pick what the database wants to pick for that. But there is an `ALTER TABLE` statement you can make to change what the next autoincrement will be.

Looks to be something like `ALTER TABLE assets AUTO_INCREMENT = 508;`

**Comment by snipe** [comment]
Changing asset ID numbers will very likely break your database though, as any actionlogs associated with those IDs will now have broken relationships. We use those asset IDs in all of our database relationship joins.

**Comment by DanCharon** [comment]
Hi Everyone,

Thank you for the swift replies. 

Ah right, well given that information I think it best that we find another way around the issue rather than altering the ID increment. 

If our next ID is 534 and the next auto increment is 540, what would be the best way to make them match? If I created a series of dummy assets and then deleted them until I hit ID 540 would that work?

Cheers!

**Reply by boring10** [reply]
@DanCharon  Are you creating your own QR codes, is that why you want them to match?

If that is the case, you can use this to go directly to an asset by the asset tag instead https://domain/hardware/bytag?assetTag=asset_tag_here.

**Reply by DanCharon** [reply]
Hey @boring10  

Yes exactly that. We've had them printed out by a third party so we have asset stickers all the way up to 1500. So now it's a case of tagging what we have and then changing their ASSET ID to match the tag in Snipe. 

The problem we've run into is that through one reason or another there is now an offset between the next ID which the stickers have in the QR code, and the next ID that appears in the list. For example:

The next unused ID is 534, however we have physically tagged assets up to 539 (these are tamper proof so sadly cannot be removed. So we can't simply swap the labels).  

So what I need to do is make enough assets so the ID catches up to 540 which is also the next unique asset ID, and also the next physical sticker we have.
![Screenshot 2021-09-30 at 12 56 58](https://user-images.githubusercontent.com/91482229/135450846-9e0353bc-300f-4bce-adda-c352d22ab396.png)

**Reply by boring10** [reply]
@DanCharon It sounds like you may be able to get away with what @uberbrady recommended and change the `AUTO_INCREMENT = 540`. It's not really a recommended thing to do since it would be easy to cause problems in the future unless your really know what you are doing.

Are you auto incrementing your asset tags or are you manually entering them? If you set your asset tags to be automatically incremented then it may help your issue from here on out. You can do that by going to Settings => Asset Tags. If you adjust your asset ID to be 540 and your next auto-increment to be 540 with a prefix of 'ASSET' then that should help to keep them in sync.

**Reply by DanCharon** [reply]
@boring10 Thank you for the help i'll give that a shot and see wher eI get up to, We've got a backup of snipe prior to any changes so rollback won't be a problem. 

The Asset tags are auto incremented at the moment, but I had toyed around with manual implementation and just reducing the number that the auto increment begins at. Which works fine, but as @snipe said, the QR codes don't care for the asset IDs so that's more of a product of the problem than the problem itself. But thank you for bringing that up, I'll put that to the team and see if they would prefer it one way or the other.

Thank you again for the help so far!
